airbyte/_future_cdk/catalog_manager.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+"""A SQL Cache implementation."""
+
+from __future__ import annotations
+
+import abc
+from typing import TYPE_CHECKING, Any, final
+
+from sqlalchemy import Column, String
+from sqlalchemy.ext.declarative import declarative_base
+
+from airbyte import exceptions as exc
+
+
+if TYPE_CHECKING:
+    from airbyte_protocol.models import (
+        ConfiguredAirbyteCatalog,
+        ConfiguredAirbyteStream,
+    )
+
+
+STREAMS_TABLE_NAME = "_airbyte_streams"
+STATE_TABLE_NAME = "_airbyte_state"
+
+GLOBAL_STATE_STREAM_NAMES = ["_GLOBAL", "_LEGACY"]
+
+Base = declarative_base()
+
+
+class CachedStream(Base):  # type: ignore[valid-type,misc]
+    __tablename__ = STREAMS_TABLE_NAME
+
+    stream_name = Column(String)
+    source_name = Column(String)
+    table_name = Column(String, primary_key=True)
+    catalog_metadata = Column(String)
+
+
+class CatalogManagerBase(abc.ABC):
```

Comment:
would it make sense to name this class catalog_manager_base.py. I got confused earlier since there is the catalog manager that inherits from it as well. Although when we move it out to the cdk, should not be an issue.